test/examples/src/lib/functions.md

```diff
@@ -116,6 +116,16 @@ d = 1
 println(d)
 ```
 
+Test setup function
+
+```@setup testsetup
+w = 5
+```
+
+```@example testsetup
+@assert w === 5
+```
+
```

Comment:
`@setup` should also work with `@repl` blocks, correct? Might be worth adding a `@repl` block test as well.